src/QueryBuilder/TableFilterQueryBuilder.php

```diff
@@ -0,0 +1,320 @@
+<?php
+
+namespace Keboola\StorageDriver\Teradata\QueryBuilder;
+
+use DateTime;
+use DateTimeZone;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Doctrine\DBAL\Query\QueryException;
+use Google\Protobuf\Internal\RepeatedField;
+use Keboola\Datatype\Definition\BaseType;
+use Keboola\Datatype\Definition\Teradata;
+use Keboola\StorageDriver\Command\Info\TableInfo;
+use Keboola\StorageDriver\Command\Table\ImportExportShared\DataType;
+use Keboola\StorageDriver\Command\Table\ImportExportShared\TableWhereFilter;
+use Keboola\StorageDriver\Command\Table\ImportExportShared\TableWhereFilter\Operator;
+use Keboola\StorageDriver\Command\Table\PreviewTableCommand;
+use Keboola\StorageDriver\Command\Table\PreviewTableCommand\PreviewTableOrderBy;
+use Keboola\StorageDriver\Command\Table\PreviewTableCommand\PreviewTableOrderBy\Order;
+use Keboola\StorageDriver\Shared\Utils\ProtobufHelper;
+use Keboola\TableBackendUtils\Escaping\Teradata\TeradataQuote;
+use LogicException;
+
+class TableFilterQueryBuilder
+{
+    public const OPERATOR_SINGLE_VALUE = [
+        Operator::eq => '=',
+        Operator::ne => '<>',
+        Operator::gt => '>',
+        Operator::ge => '>=',
+        Operator::lt => '<',
+        Operator::le => '<=',
+    ];
+    public const OPERATOR_MULTI_VALUE = [
+        Operator::eq => 'IN',
+        Operator::ne => 'NOT IN',
+    ];
+
+    public const DEFAULT_CAST_SIZE = 16384;
+
+    private Connection $connection;
+    private ?TableInfo $tableInfo;
+    private ColumnConverter $columnConverter;
+
+    public function __construct(
+        Connection $connection,
+        ?TableInfo $tableInfo,
+        ColumnConverter $columnConverter
+    ) {
+        $this->connection = $connection;
+        $this->tableInfo = $tableInfo;
+        $this->columnConverter = $columnConverter;
+    }
+
+    public function buildQueryFromCommand(
+        PreviewTableCommand $options,
+        string $schemaName
+    ): TableFilterQueryBuilderResponse {
+        $this->assertFilterCombination($options);
+
+        $query = new QueryBuilder($this->connection);
+
+        $this->processChangedConditions($options, $query);
+
+        try {
+            if ($options->getFulltextSearch() !== '') {
+                if ($this->tableInfo === null) {
```

Comment:
tohle je divny, asi to tak nechame, ale radeji bych to videl nejak rozdelene.

Reply:
Nevim jak to udelat jinak...
Potrebuju sem dostat vsechny sloupce dany tabulky - v Connection se to vytahuje z Metadat, tady se musi pres InfoTable. A je to potreba jen pro `fulltextSearch`... 🤷